### ISSUE TITLE
Shutdown hook errors if terminal stayed paused

### ISSUE THREAD

**Issue by bdw429s** [jline/jline3#462]
As a result of this ticket https://github.com/jline/jline3/issues/383 I'm trying to create the terminal paused and only resume it if my console app actually needs to read from the standard input.  This prevents JLine from gobbling up text on the standard input that wasn't intended for it.

However, the `AbstractWindowsTerminal` shutdown hook doesn't take into account the possibility that the pump object is null so an exception gets thrown when exiting the app.

```
Nov 08, 2019 11:33:36 PM org.jline.utils.Log logr
WARNING: Task failed
java.lang.NullPointerException
        at org.jline.terminal.impl.AbstractWindowsTerminal.close(AbstractWindowsTerminal.java:219)
        at org.jline.utils.ShutdownHooks.runTasks(ShutdownHooks.java:55)
        at org.jline.utils.ShutdownHooks.access$000(ShutdownHooks.java:21)
        at org.jline.utils.ShutdownHooks$1.run(ShutdownHooks.java:36)
```

It looks like the `close()` method needs to check `pump` and only try to interrupt it if it's not null.

**Comment by mattirn** [comment]
Fixed in 3.13.1,  #442

**Comment by bdw429s** [comment]
@mattirn Awesome, thanks for the link.  I actually thought I was using 3.13, but it looks like I had reverted back to 3.12 while testing.  I can confirm updating to 3.13 has fixed this!
